src/plugins/vis_type_vega/public/expressions/helpers.test.js

```diff
@@ -0,0 +1,199 @@
+      expect(
+        JSON.stringify(
+          createSpecFromDatatable(
+            formatDatatable(JSON.parse(datatable)),
+            JSON.parse(visParams),
+            JSON.parse(dimensions)
+          )
+        )
+      ).toBe(spec);
```

Comment:
Not sure `toMatchSnapshot` makes sense in this context? There is nothing being rendered; just a return value from a helper fn.

Also, would moving those to a `__mocks__` make sense here? There is no mocked values being done here, rather just some value constants used as input/output to the helper fn.

Reply:
au contrarie, this is exactly what snapshots are really good for. The official docs suggest to use them for ui, but they are essentially ensuring that structure of the value that you are expecting does not change between runs. And the best part it that it can also point out where the change did occur which the `toBe` function will not tell you. That being said this is a nit so i'm okay with it going in as is too :) 

Yeah moving them to `__mocks__` would be nice too, but totally optional!

Reply:
Ah got it, I see how that can work now. Thanks for the suggestion!

Reply:
Updated in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3145/commits/9c2e3496a8dbfbcbb41a4ae27671e5c0f9c4603f